SubList function documentation (#9718)
opensearch-project/documentation-website: _data-prepper/pipelines/sublist.md

---
layout: default
title: subList()
parent: Functions
grand_parent: Pipelines
nav_order: 35
---

# subList(<key>, <start_index, inclusive>, <end_index, exclusive>)

The `subList()` function extracts a sublist from a list field in an event. It takes the following arguments:

> - A JSON pointer to a list field in the event

Reviewer: ```suggestion
- A JSON pointer to an event field containing a list
```

- A start index (inclusive)
- An end index (exclusive)

The function returns the portion of the list between the specified start and end indexes. If the end index is `-1`, the function extracts elements from the start index to the end of the list.

## Examples

The following examples show how the `sublist()` function works.

### add_entries processor

You can use `subList()` in the `add_entries` processor, as shown in the following example.

> For example, the function will use the following input to extact a sublist:

Reviewer: ```suggestion
The function uses the following input to extract a sublist:
```

```
input: {"my_list": [ 0, 1, 2, 3, 4, 5, 6]}
```

Then, the following configuration uses the `add_entries` processor to extract a sublist from `my_list`, starting at index 1 and ending before index 4: 

```yaml
add_entries:
  entries:
    - key: "my_list"
      value_expression: '/subList(/my_list, 1, 4)'
      overwrite_if_key_exists: true
```

> {% include copy-curl.html %}

Reviewer: ```suggestion
{% include copy.html %}
```
